features/dd-sdk-android-flags/src/main/kotlin/com/datadog/android/flags/featureflags/FlagsClient.kt

```diff
@@ -52,12 +52,14 @@ object FlagsClient {
         return synchronized(registeredProviders) {
             val provider = registeredProviders[sdkCore]
             if (provider == null) {
+                val errorMsg = "No FlagsProvider for the SDK instance with name ${sdkCore.name} " +
+                    "found, returning no-op implementation."
                 (sdkCore as? FeatureSdkCore)
                     ?.internalLogger
                     ?.log(
                         InternalLogger.Level.WARN,
                         InternalLogger.Target.USER,
-                        { NO_PROVIDER_REGISTERED_MESSAGE.format(Locale.US, sdkCore.name) }
```

Comment:
ooc, what what is the detekt rule that is against what looks like a pretty clean string build?

Reply:
UnsafeThirdPartyFunctionCall (format potentially throwing IllegalFormatException)